### ISSUE TITLE
Use reflinks in mkfs.btrfs when using --rootdir?

### ISSUE THREAD

**Issue by DaanDeMeyer** [kdave/btrfs-progs#547]
I noticed that when using mkfs.btrfs on a loopback file with --rootdir, it doesn't seem to try to reflink the files into the filesystem. Would this be possible at all to avoid copying the data?

**Comment by kdave** [comment]
Yeah reflink is not implemented. It could as long as it would work for the source and target. It's in `add_file_items`, source file is read by sectors and needs the checksum calculated. Reflinking would need to look for longest contiguous extent while still adding the checksums.

**Comment by kdave** [comment]
Rootdir + reflink was implemented in #1023 and released in 6.16.1.
